### PR TITLE
feat: wait for node to be ready option

### PR DESCRIPTION
**Pull request by shuplenkov**
## Issue being fixed or feature implemented

To implement CI we need to get acknowledgment that node is ready to accept connections.

## What was done?

- Introduced `wait-for-readiness` option for `start` command
- Introduced `wait-for-readiness` option for `group:start` command
- Fixed local network setup issues

## How Has This Been Tested?

Running commands

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: src/commands/group/start.js [hunk 1]
```diff
@@ -53,22 +53,8 @@ class GroupStartCommand extends GroupBaseCommand {
...
-                success = !response.error;
-              }
-
-              if (!success) {
-                await wait(2000);
-              }
-            } while (!success);
-          },
+          skip: !waitForTenderdash,
+          task: () => waitForTenderdashTask(configGroup[configGroup.length - 2]),
```
Comment (shumkov) [inline src/commands/group/start.js:57]: Wait for all nodes except nodes without platform.

File: src/commands/group/start.js [hunk 2]
```diff
@@ -53,22 +53,8 @@ class GroupStartCommand extends GroupBaseCommand {
...
-              if (response) {
-                success = !response.error;
-              }
-
-              if (!success) {
-                await wait(2000);
-              }
-            } while (!success);
-          },
+          skip: !waitForTenderdash,
```
Comment (shumkov) [inline src/commands/group/start.js:56]: enabled

File: src/commands/group/start.js [hunk 3]
```diff
@@ -53,22 +53,8 @@ class GroupStartCommand extends GroupBaseCommand {
         },
         {
           title: 'Wait for Tenderdash',
```
Comment (shumkov) [inline src/commands/group/start.js:55]: Await for nodes to be ready

File: src/commands/group/start.js [hunk 4]
```diff
@@ -22,12 +21,13 @@ class GroupStartCommand extends GroupBaseCommand {
...
       'drive-image-build-path': driveImageBuildPath,
       'dapi-image-build-path': dapiImageBuildPath,
+      'wait-for-tenderdash': waitForTenderdash,
       verbose: isVerbose,
     },
     dockerCompose,
     startNodeTask,
     configGroup,
-    createTenderdashRpcClient,
+    waitForTenderdashTask,
```
Comment (shumkov) [inline src/commands/group/start.js:30]: waitForNodeToBeReadyTask

File: src/commands/group/start.js [hunk 5]
```diff
@@ -96,6 +82,7 @@ GroupStartCommand.flags = {
   update: flagTypes.boolean({ char: 'u', description: 'download updated services before start', default: false }),
   'drive-image-build-path': flagTypes.string({ description: 'drive\'s docker image build path', default: null }),
   'dapi-image-build-path': flagTypes.string({ description: 'dapi\'s docker image build path', default: null }),
+  'wait-for-tenderdash': flagTypes.boolean({ description: 'wait for tenderdash to start', default: false }),
```
Comment (shumkov) [inline src/commands/group/start.js:85]: `wait-for-readiness`

File: src/listr/tasks/setup/local/initializePlatformTaskFactory.js [hunk 6]
```diff
@@ -53,8 +47,8 @@ function initializePlatformTaskFactory(
         },
       },
       {
-        title: 'Wait 20 seconds to ensure all services are running',
-        task: () => wait(20000),
+        title: 'Wait for tenderdash to start',
```
Comment (shumkov) [inline src/listr/tasks/setup/local/initializePlatformTaskFactory.js:50]: Await for nodes to be ready

File: src/listr/tasks/setup/local/initializePlatformTaskFactory.js [hunk 7]
```diff
@@ -53,8 +47,8 @@ function initializePlatformTaskFactory(
...
+        task: () => waitForTenderdashTask(configGroup[configGroup.length - 2]),
```
Comment (shumkov) [inline src/listr/tasks/setup/local/initializePlatformTaskFactory.js:51]: All nodes, except without platform

File: src/listr/tasks/setup/local/activateSporksTaskFactory.js [hunk 8]
```diff
@@ -49,8 +49,9 @@ function activateSporksTaskFactory(
         title: 'Waiting for quorums',
```
Comment (shumkov) [inline src/listr/tasks/setup/local/activateSporksTaskFactory.js:49]: Move out to initializePlatformTask

File: src/commands/group/start.js [hunk 9]
```diff
@@ -4,7 +4,6 @@ const { flags: flagTypes } = require('@oclif/command');
 
 const GroupBaseCommand = require('../../oclif/command/GroupBaseCommand');
 const MuteOneLineError = require('../../oclif/errors/MuteOneLineError');
-const wait = require('../../util/wait');
 
 class GroupStartCommand extends GroupBaseCommand {
```
Comment (shumkov) [inline src/commands/group/start.js:8]: Let's implement this flag for `mn start` too

File: src/core/waitForCoreQuorum.js [hunk 10]
```diff
@@ -1,13 +1,18 @@
...
 /**
  * Wait Core to set quorum
  *
  * @typedef {waitForCoreQuorum}
  * @param {RpcClient} rpcClient
+ * @param {string} network
  * @return {Promise<void>}
  */
-async function waitForCoreQuorum(rpcClient) {
+async function waitForCoreQuorum(rpcClient, network) {
```
[the remark this answers is not in the document]
Comment (shuplenkov) [reply src/core/waitForCoreQuorum.js:12]: Why enable? We just wait here

**Review by shumkov — APPROVED** [review]
Looks good! 👍

File: src/core/waitForCoreQuorum.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (shumkov) [reply src/core/waitForCoreQuorum.js]: We generating blocks to enable, right?

**Pull request merged**